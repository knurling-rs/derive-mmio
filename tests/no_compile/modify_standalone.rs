
#[derive(derive_mmio::Mmio)]
#[repr(C)]
struct Uart {
    // Standalone modification is not allowed, Read and Write access need to be specified
    // explicitely.
    #[mmio(Modify)]
    fifo: u32,
}

fn main() {
    let mut uart = Uart { fifo: 0x0 };

    // Safety: We're pointing at a real object
    let mut mmio_uart = unsafe { Uart::new_mmio(core::ptr::addr_of_mut!(uart)) };
    mmio_uart.modify_uart();
}
