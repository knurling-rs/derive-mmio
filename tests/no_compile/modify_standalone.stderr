error: Detected Modify field attribute without read and/or write access specifiers
 --> tests/no_compile/modify_standalone.rs:7:5
  |
7 |     #[mmio(Modify)]
  |     ^

error[E0599]: no function or associated item named `new_mmio` found for struct `Uart` in the current scope
  --> tests/no_compile/modify_standalone.rs:15:40
   |
4  | struct Uart {
   | ----------- function or associated item `new_mmio` not found for this struct
...
15 |     let mut mmio_uart = unsafe { Uart::new_mmio(core::ptr::addr_of_mut!(uart)) };
   |                                        ^^^^^^^^ function or associated item not found in `Uart`
